var/spack/repos/builtin/packages/flann/package.py

```diff
@@ -0,0 +1,134 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+
+
+class Flann(CMakePackage):
+    """
+    FLANN is a library for performing fast approximate nearest neighbor
+    searches in high dimensional spaces. It contains a collection of
+    algorithms we found to work best for nearest neighbor search and a system
+    for automatically choosing the best algorithm and optimum parameters
+    depending on the dataset.
+
+    FLANN is written in C++ and contains bindings for the following languages:
+    C, MATLAB and Python.
+
+    NOTE: The author of the package was careful to support HDF5 without
+    parallel support, so the dependencies do not require this, but you will
+    likely want to make sure that you are giving hdf5+mpi as the dependency,
+    this is the spack default so it should be fine.
```

Comment:
I don't think this NOTE is necessary. This is the case for pretty much every package in Spack that comes with optional dependencies. As long as the variant defaults are reasonable, it's up to the user to explicitly deactivate this optional support.